specs/interop/shared-locbox.md

```diff
@@ -0,0 +1,300 @@
+**Table of Contents**
+
+- [Overview](#overview)
+- [Implementation](#implementation)
+  - [`SharedLockbox`](#sharedlockbox)
+  - [`SuperchainConfig`](#superchainconfig)
+  - [`OptimismPortal`](#optimismportal)
+  - [`LiquidityMigrator`](#liquiditymigrator)
+  - [Future Considerations / Additional Notes](#future-considerations--additional-notes)
+- [Upgrade and migration process](#upgrade-and-migration-process)
+  - [Batch transaction process](#batch-transaction-process)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+With interoperable ETH, withdrawals may fail if the referenced `OptimismPortal` lacks sufficient ETH
+— making it mandatory to withdraw from the L2 where the ETH was originally deposited.
+To improve the Superchain's interoperable ETH withdrawal user experience and avoid this issue,
+the `SharedLockbox` serves as a solution.
+It unifies ETH L1 liquidity in a single contract, enabling seamless withdrawals of ETH from any OP chain in the Superchain,
+regardless of where the ETH was initially deposited.
+
+## Implementation
+
+### `SharedLockbox`
+
+The `SharedLockbox` contract is designed to manage the unified ETH liquidity for the Superchain.
+It provides two main functions: `lockETH` for depositing ETH into the lockbox,
+and `unlockETH` for withdrawing ETH from the lockbox.
+These functions are called by the `OptimismPortal` contracts to manage the shared ETH liquidity
+when making deposits or finalizing withdrawals.
+
+**Storage layout**
+
+The `SharedLockbox` contract maintains a mapping of authorized `OptimismPortal` addresses.
+This ensures that only approved portals can interact with the shared liquidity pool.
+
+```solidity
+// OptimismPortals that are part of the dependency cluster.
+mapping(address _portal => bool) internal _authorizedPortals;
+```
+
+**Interface**
+
+The `ShareLockbox` interface is the following:
+
+```solidity
+function lockETH() external payable;
+
+function unlockETH(uint256 _value) external;
+
+function authorizePortal(address _portal) external;
+
+event ETHLocked(address indexed portal, uint256 amount);
+
+event ETHUnlocked(address indexed portal, uint256 amount);
+
+event AuthorizedPortal(address indexed portal);
+```
+
+A possible implementation could look like this:
+
+```solidity
+function lockETH() external payable {
+    require(_authorizedPortals[msg.sender], "Unauthorized");
+
+    emit ETHLocked(msg.sender, msg.value);
+}
+
+function unlockETH(uint256 _value) external {
+    require(_authorizedPortals[msg.sender], "Unauthorized");
+
+    // Using `donateETH` to not trigger a deposit
+    IOptimismPortal(msg.sender).donateETH{ value: _value }();
+
+    emit ETHUnlocked(msg.sender, _value);
+}
+
+function authorizePortal(address _portal) external {
+    require(msg.sender == superchainConfig, "Unauthorized");
+
+    _authorizedPortals[_portal] = true;
+
+    emit AuthorizedPortal(_portal);
+}
+```
+
+### `SuperchainConfig`
```

Comment:
I would describe this contract before the shared lockbox, as it feels functional to explaining the lockbox